### PR TITLE
Add ReduceMerge transformation

### PR DESCRIPTION
**Pull request by artkuli**
### Tickets:
 - *60918*

### CODE REVIEW

**Review by mbencer — APPROVED** [review]
LGTM! Please consider changing at least one test to compare the produced function with expected instead of counting ops.

**Comment by mbencer** [comment]
Thanks for test improvements ;) No more comments

File: src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    auto input = pattern::any_input();
+    auto first_axis = pattern::any_input();
+    auto reduce = pattern::wrap_type<T>({input, first_axis});
+    auto second_axis = pattern::any_input();
+    return pattern::wrap_type<T>({reduce, second_axis});
+}
+
+template <typename T>
+bool fuse_reduce_operations(const std::shared_ptr<Node>& node) {
+    const auto bottom_reduce = std::dynamic_pointer_cast<T>(node);
```
Comment (mateusztabaka) [inline src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp:28]:
```suggestion
    const auto bottom_reduce = as_type_ptr<T>(node);
```

File: src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                    return first_axis > second_axis;
+                });
+            })) {
+            return false;
+        }
+    }
+
+    auto concat_axes =
+        std::make_shared<opset9::Concat>(OutputVector{top_reduce->input_value(1), bottom_reduce->input_value(1)},
+                                         int64_t(0));
```
Comment (mateusztabaka) [inline src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp:58]: please call get_constant_from_source on concat_axes to constantfold the axes

Comment (mateusztabaka) [inline src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp:58]: please call get_constant_from_source on concat_axes to constantfold the axes

**Review by mateusztabaka — COMMENTED** [review]
Some comments on tests:
- please enable ACCURACY and CONST_VALUES in comparator
- add some tests with more than 2 Reduces in a row
- use input shape other than {1}
-  add a test with input with dynamic shape

File: src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp (hunk 2 above)
Comment (artkuli) [reply src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp:59]: Fixed

File: src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp (hunk 1 above)
Comment (artkuli) [reply src/common/transformations/src/transformations/common_optimizations/reduce_merge.cpp]: Fixed

**Comment by artkuli** [comment]
> Some comments on tests:
> 
> * please enable ACCURACY and CONST_VALUES in comparator
> * add some tests with more than 2 Reduces in a row
> * use input shape other than {1}
> * add a test with input with dynamic shape

Done

**Review by mateusztabaka — APPROVED** [review]

**Pull request merged**